core/src/main/java/com/google/errorprone/bugpatterns/UnusedMethod.java

```diff
@@ -78,23 +78,33 @@
 
   private static final ImmutableSet<String> EXEMPTING_METHOD_ANNOTATIONS =
       ImmutableSet.of(
+          "com.fasterxml.jackson.annotation.JsonCreator",
           "com.google.inject.Provides",
           "com.google.inject.Inject",
           "com.google.inject.multibindings.ProvidesIntoMap",
           "com.google.inject.multibindings.ProvidesIntoSet",
           "javax.annotation.PreDestroy",
           "javax.annotation.PostConstruct",
-          "javax.persistence.PostLoad",
           "javax.inject.Inject",
-          "org.testng.annotations.DataProvider");
+          "javax.persistence.PostLoad",
+          "org.aspectj.lang.annotation.Pointcut",
+          "org.aspectj.lang.annotation.Before",
+          "org.springframework.context.annotation.Bean",
+          "org.testng.annotations.AfterClass",
+          "org.testng.annotations.AfterMethod",
+          "org.testng.annotations.BeforeClass",
+          "org.testng.annotations.BeforeMethod",
+          "org.testng.annotations.DataProvider",
+          "org.junit.AfterClass",
+          "org.junit.BeforeClass");
 
   /** The set of types exempting a type that is extending or implementing them. */
   private static final ImmutableSet<String> EXEMPTING_SUPER_TYPES = ImmutableSet.of();
 
   @Override
   public Description matchCompilationUnit(CompilationUnitTree tree, VisitorState state) {
-    // Map of symbols to method declarations. Initially this is a map of all of the methods. As we
-    // go we remove those variables which are used.
+    // Map of symbols to method declarations. Initially this is a map of all symbols and paths of
+    // the methods. As we go we remove those variables which are used.
```

Comment:
Not sure this change really clarifies things. In order to increase the likelihood of getting this PR merged upstream I propose to keep this change out.